data-conversion/src/main.rs

```diff
@@ -1,104 +1,174 @@
 //use std::env;
 use argh::FromArgs;
+use std::error::Error;
+use std::fmt;
 use std::fs::read_to_string;
 use std::fs::File;
 use std::io::{self, Write};
+use std::str::FromStr;
 
-fn main() {
-    #[derive(FromArgs)]
-    /// get arguments to convert
-    struct Arguments {
-        /// file to convert from
-        #[argh(option)]
-        from: String,
-
-        /// file to convery to
-        #[argh(option)]
-        to: String,
-
-        /// type to convert from
-        #[argh(option)]
-        ftype: String,
-
-        /// type to convert to
-        #[argh(option)]
-        totype: String,
+//cargo run -- --from $PATH1 --to $PATH2 --ftype "binary" --totype "hex"
+
+#[derive(Debug)]
+struct ParseNumTypeError;
+
+impl fmt::Display for ParseNumTypeError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "invalid number type")
     }
+}
 
-    let args: Arguments = argh::from_env();
+impl Error for ParseNumTypeError {}
 
-    // Array of all possible options to convert from
-    let from_types: Vec<String> = vec![
-        "binary".to_string(),
-        "float".to_string(),
-        "hex".to_string(),
-        "fixed".to_string(),
-    ];
-    // Array of all possible options to convert to
-    let to_types: Vec<String> =
-        vec!["binary".to_string(), "float".to_string(), "hex".to_string()];
-
-    if !from_types.contains(&args.ftype) {
-        panic!("{} is not a valid type to convert from", args.from);
+#[derive(Debug, PartialEq)] // Add PartialEq derivation here - What is this?
```

Comment:
To answer the question: deriving `PartialEq` lets you compare values of this type with `==`. So if variables `x` and `y` are of type `NumType`, now you can do `x == y` to see if they're the same.